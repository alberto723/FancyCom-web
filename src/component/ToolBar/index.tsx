import React, { useState } from 'react';
import FormatAlignJustifyIcon from '@material-ui/icons/FormatAlignJustify';
import SearchIcon from '@material-ui/icons/Search';
import PersonAddOutlinedIcon from '@material-ui/icons/PersonAddOutlined';
import ExpandMoreOutlinedIcon from '@material-ui/icons/ExpandMoreOutlined';
import logo from './ambid_logo.png';
import user_img from './ambid_user.png';
// import './style.css';
import { makeStyles } from '@material-ui/core/styles';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import { Button } from '@material-ui/core';

export interface ToolBarProps {
  name?: string;
  leftItems?: object[];
  title: string;
  rightItems?: object[];
}

export const ToolBar: React.FC<ToolBarProps> = ({ name, leftItems, title, rightItems }) => {
  const styles: any = useStyles();
  const [titleModalVisible, setTitleModalVisible] = useState(false);
  const [titleText, setTitleText] = useState(title);

  const handleClose = () => {
    setTitleModalVisible(false);
  };

  const onChangeTitleText = (event: any) => {
    setTitleText(event.target.value);
  };

  const renderTitleEditModal = (
    <Dialog
      open={titleModalVisible}
      onClose={handleClose}
      aria-labelledby='alert-dialog-title'
      aria-describedby='alert-dialog-description'>
      <DialogTitle id='alert-dialog-title'>ToolBar Title</DialogTitle>
      <DialogContent>
        <div className={styles.centeredView}>
          <textarea
            className={styles.modalEdiTitleInput}
            onChange={onChangeTitleText}
            value={titleText}
            placeholder='Title'
            maxLength={30}
          />
        </div>
      </DialogContent>
      <DialogActions>
        <Button onClick={() => setTitleModalVisible(false)} color='primary' autoFocus>
          OK
        </Button>
      </DialogActions>
    </Dialog>
  );

  return (
    <div className={styles.screen}>
      <div className={styles.container}>
        <div className={styles.leftItemsSection}>
          <div>
            <FormatAlignJustifyIcon style={{ fontSize: 30, color: '#9EA9B1' }} />
          </div>
          <img src={logo} alt='Logo' className={styles.logoImg} />
        </div>
        <div onClick={() => setTitleModalVisible(true)}>
          <p className={styles.toolbarTitleText}>{titleText}</p>
        </div>
        <div>
          <div className={styles.rightItemsSection}>
            <div className={styles.rightItemElement}>
              <SearchIcon style={{ fontSize: 30, color: '#9EA9B1' }} />
            </div>
            <div className={styles.rightItemElement}>
              <PersonAddOutlinedIcon style={{ fontSize: 35, color: '#9EA9B1' }} />
            </div>
            <div className={styles.publishBtn}>
              <p className={styles.publishBtnText}>PUBLISH</p>
            </div>
            <div className={styles.userAvatarPart}>
              <img src={user_img} alt='User' className={styles.userAvatarImg} />
              <ExpandMoreOutlinedIcon style={{ fontSize: 25, color: '#9EA9B1' }} />
            </div>
          </div>
        </div>
      </div>
      {titleModalVisible && renderTitleEditModal}
    </div>
  );
};

const useStyles = makeStyles(() => ({
  screen: {
    flex: 1,
    backgroundColor: 'white',
    paddingTop: '10px',
  },
  scrollViewWrapper: {
    display: 'flex',
    borderBottomWidth: '2px',
    borderBottomColor: '#E7EEF2',
    borderTopWidth: '2px',
    borderTopColor: '#E7EEF2',
    flexDirection: 'row'
  },
  container: {
    display: 'flex',
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: '10px',
    paddingRight: '10px',
  },
  leftItemsSection: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  logoImg: {
    marginLeft: '5px',
    marginRight: '5px',
  },
  toolbarTitleText: {
    color: '#607380',
    paddingLeft: '15px',
    paddingRight: '15px',
  },
  rightItemsSection: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center'
  },
  publishBtn: {
    backgroundColor: '#F35564',
    paddingLeft: '20px',
    paddingRight: '20px',
    borderRadius: '20px',
    marginLeft: '5px',
    marginRight: '5px',
  },
  publishBtnText: {
    color: 'white'
  },
  userAvatarPart: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  userAvatarImg: {
    width: '40px',
    height: '40px',
    marginLeft: '5px',
    marginRight: '5px',
  },
  rightItemElement: {
    marginLeft: '5px',
    marginRight: '5px',
  },
  centeredView: {
    display: 'flex',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: '22px',
    marginLeft: '30px',
    marginRight: '30px',
    minWidth: '350px',
  },
  modalView: {
    margin: '20px',
    backgroundColor: 'white',
    borderRadius: '20px',
    padding: '35px',
    alignItems: 'center',
    width: '100%',
  },
  closeButton: {
    backgroundColor: '#2196F3',
    borderRadius: '20px',
    padding: '10px',
    width: '100%',
    marginTop: '15px',
  },
  textStyle: {
    color: 'white',
    fontWeight: 'bold',
    textAlign: 'center',
  },
  modalTitle: {
    marginBottom: '15px',
    fontSize: '20px',
    fontWeight: 'bold',
  },
  modalEdiTitleInput: {
    width: '100%',
    height: '40px',
    borderColor: 'grey',
    borderWidth: '1px',
    paddingLeft: '10px',
    paddingRight: '10px',
    borderRadius: '10px',
  }
}));
