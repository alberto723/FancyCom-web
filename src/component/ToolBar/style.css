.screen {
  flex: 1;
  background-color: white;
  padding-top: 10px;
}
.scrollViewWrapper {
  display: flex;
  border-bottom-width: 2px;
  border-bottom-color: #E7EEF2;
  border-top-width: 2px;
  border-top-color: #E7EEF2;
  flex-direction: row;
}
.container {
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;
  padding-left: 10px;
  padding-right: 10px;
}
.leftItemsSection {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}
.logoImg {
  margin-left: 5px;
  margin-right: 5px;
}
.toolbarTitleText {
  color: #607380;
  padding-left: 25px;
  padding-right: 25px;
}
.rightItemsSection {
  display: flex;
  flex-direction: 'row';
  align-items: center;
  justify-content: center;
}
.publishBtn {
  background-color: #F35564;
  padding-left: 20px;
  padding-right: 20px;
  border-radius: 20px;
  margin-left: 5px;
  margin-right: 5px;
}
.publishBtnText {
  color: white;
}
.userAvatarPart {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}
.userAvatarImg {
  width: 40px;
  height: 40px;
  margin-left: 5px;
  margin-right: 5px;
}
.rightItemElement {
  margin-left: 5px;
  margin-right: 5px;
}
.centeredView {
  display: flex;
  flex: 1;
  justify-content: center;
  align-items: center;
  margin-top: 22px;
  margin-left: 30px;
  margin-right: 30px;
  min-width: 350px;
}
.modalView {
  margin: 20px;
  background-color: white;
  border-radius: 20px;
  padding: 35px;
  align-items: center;
  /* shadowColor: "#000";
  shadowOffset: {
    width: 0;
    height: 2
  };
  shadowOpacity: 0.25;
  shadowRadius: 3.84;
  elevation: 5; */
  width: 100%;
}
.closeButton {
  background-color: #2196F3;
  border-radius: 20px;
  padding: 10px;
  /* elevation: 2; */
  width: 100%;
  margin-top: 15;
};
.textStyle {
  color: white;
  font-weight: bold;
  text-align: center;
};
.modalTitle {
  margin-bottom: 15px;
  font-size: 20px;
  font-weight: bold;
}
.modalEdiTitleInput {
  width: 100%;
  height: 40px;
  border-color: grey;
  border-width: 1px;
  padding-left: 10px;
  padding-right: 10px;
  border-radius: 10px;
}
