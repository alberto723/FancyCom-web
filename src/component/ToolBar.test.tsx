// Canvas.test.js
import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';

import { ToolBar } from './ToolBar';

test('Should be rendered in the dom', () => {
  const component = renderer.create(
    <ToolBar
      title='Project Name > Object Name'
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should show the same number of items in the left side of the toolbar as the number of items in the leftItems prop', () => {
  const { queryByLabelText, getByText, container } = render(
    <ToolBar
      title='Project Name > Object Name'
    />
  );

  expect(container.querySelector('.makeStyles-leftItemsSection-4')?.innerHTML != null ).toBe(true);
});

test('Should show the same number of items in the right side of the toolbar as the number of items in the rightItems prop', () => {
  const { queryByLabelText, getByText, container } = render(
    <ToolBar
      title='Project Name > Object Name'
    />
  );

  expect(container.querySelector('.makeStyles-rightItemsSection-7')?.innerHTML != null ).toBe(true);
});

test('Should show a title if one is provided', () => {
  const { queryByLabelText, getByText, container } = render(
    <ToolBar
      title='Project Name > Object Name'
    />
  );

  expect(container.querySelector('.makeStyles-toolbarTitleText-6')?.innerHTML).toContain('Project Name');
});

test('Should show a child element if one is provided', () => {
  const { queryByLabelText, getByText, container } = render(
    <ToolBar
      title='Project Name > Object Name'
    />
  );

  expect(container.querySelector('.makeStyles-screen-1')?.innerHTML != null).toBe(true);
});
