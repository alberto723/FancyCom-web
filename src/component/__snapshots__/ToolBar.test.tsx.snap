// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be rendered in the dom 1`] = `
<div
  className="makeStyles-screen-1"
>
  <div
    className="makeStyles-container-3"
  >
    <div
      className="makeStyles-leftItemsSection-4"
    >
      <div>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          style={
            Object {
              "color": "#9EA9B1",
              "fontSize": 30,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M3 21h18v-2H3v2zm0-4h18v-2H3v2zm0-4h18v-2H3v2zm0-4h18V7H3v2zm0-6v2h18V3H3z"
          />
        </svg>
      </div>
      <img
        alt="Logo"
        className="makeStyles-logoImg-5"
        src=""
      />
    </div>
    <div
      onClick={[Function]}
    >
      <p
        className="makeStyles-toolbarTitleText-6"
      >
        Project Name &gt; Object Name
      </p>
    </div>
    <div>
      <div
        className="makeStyles-rightItemsSection-7"
      >
        <div
          className="makeStyles-rightItemElement-12"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "#9EA9B1",
                "fontSize": 30,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-rightItemElement-12"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "#9EA9B1",
                "fontSize": 35,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0-6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2zm0 8c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4zm-6 4c.22-.72 3.31-2 6-2 2.7 0 5.8 1.29 6 2H9zm-3-3v-3h3v-2H6V7H4v3H1v2h3v3z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-publishBtn-8"
        >
          <p
            className="makeStyles-publishBtnText-9"
          >
            PUBLISH
          </p>
        </div>
        <div
          className="makeStyles-userAvatarPart-10"
        >
          <img
            alt="User"
            className="makeStyles-userAvatarImg-11"
            src=""
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "#9EA9B1",
                "fontSize": 25,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6-1.41-1.41z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
